### PR TITLE
Cleanup how fields_to_plot = none is handled

### PR DESCRIPTION
**Pull request by dpgrote**
There has been a change in AMReX that allows MultiFabs to be created with ncomp = 0. (AMReX PR #2873) This allows the handling the `fields_to_plot = none` to be cleaned up, removing the `if` checks around the allocation and call to `WriteToFile`.

This also fixes another issue that arose in PR #2419, that the plot files were not readable by Yt (because the Header file was not written out). With this PR, that is also fixed.

Once the version of AMReX that WarpX depends on is updated, I remove the WIP and this can be merged.

### CODE REVIEW

File: Source/Diagnostics/FullDiagnostics.cpp [hunk 1]
```diff
@@ -540,12 +540,8 @@ FullDiagnostics::InitializeBufferData (int i_buffer, int lev ) {
     if (use_warpxba == false) dmap = amrex::DistributionMapping{ba};
     // Allocate output MultiFab for diagnostics. The data will be stored at cell-centers.
     int ngrow = (m_format == "sensei" || m_format == "ascent") ? 1 : 0;
-    // The zero is hard-coded since the number of output buffers = 1 for FullDiagnostics
```
Comment (dpgrote) [inline Source/Diagnostics/FullDiagnostics.cpp:543]: Note for reviewer - this comment was left over from previous changes and is no longer meaningful so was delete here for clean up.

**Comment by EZoni** [comment]
@dpgrote Is the version of AMReX pulled in #3211 (now merged) good enough to remove the [WIP] tag here?

**Comment by dpgrote** [comment]
@EZoni Unfortunately no. The merge was made this morning and is newer than the AMReX version in PR #3211.

**Comment by ax3l** [comment]
Feel free to bump AMReX to the latest version in the current PR via this script:
`./Tools/Release/updateAMReX.py`

**Comment by ax3l** [comment]
I went ahead and pushed the AMReX update :)

**Review by ax3l — APPROVED** [review]
LGTM, thank you :)

**Comment by dpgrote** [comment]
Thanks @ax3l!

**Pull request merged**